src/obj_char_transfer.c

```diff
@@ -0,0 +1,872 @@
+#include "global.h"
+#include "obj_char_transfer.h"
+#include "vram_transfer_manager.h"
+
+typedef enum ObjCharTransferTaskState {
+    OBJ_CHAR_TRANSFER_TASK_INIT,
+    OBJ_CHAR_TRANSFER_TASK_1,
+    OBJ_CHAR_TRANSFER_TASK_2,
+    OBJ_CHAR_TRANSFER_TASK_COPYABLE,
+    OBJ_CHAR_TRANSFER_TASK_COPY,
+} ObjCharTransferTaskState;
+
+typedef struct ObjCharTransferTask {
+    NNSG2dCharacterData *charData;
+    NNS_G2D_VRAM_TYPE vram;
+    u8 isTransferred;
+    int resId;
+    NNSG2dImageProxy imageProxy;
+    u32 baseAddrMain;
+    u32 baseAddrSub;
+    u8 state;
+    BOOL shouldUpdateMappingTypeFromHW;
+    BOOL transferFromTail;
+    BOOL needResetBlockTransferFlags;
+    u32 sizeMain;
+    u32 sizeSub;
+} ObjCharTransferTask;
+
+struct ObjCharTransferTasksManager {
```

Comment:
genuinely I'd prefer the type definition and the variable declaration to be separate, it's cleaner imo